build/build-race-binaries.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Build cockroach binary with race detection enabled.
+
+set -euo pipefail
+
+source $(dirname $0)/build-common.sh
+
+# This is mildly tricky: This script runs itself recursively. The
+# first time it is run it does not take the if-branch below and
+# executes on the host computer. It uses the builder.sh script to run
+# itself inside of docker passing "docker" as the argument causing the
+# commands in the if-branch to be executed within the docker
+# container.
+if [ "${1-}" = "docker" ]; then
+    time make STATIC=1 build GOFLAGS="-race"
+    check_static cockroach
+    strip -S cockroach
```

Comment:
Sounds reasonable to me.

On Fri, Apr 22, 2016 at 5:30 AM marc <notifications@github.com> wrote:

> In build/build-race-binaries.sh
> <https://github.com/cockroachdb/cockroach/pull/6228#discussion_r60713289>:
>
> > +# Build cockroach binary with race detection enabled.
> > +
> > +set -euo pipefail
> > +
> > +source $(dirname $0)/build-common.sh
> > +
> > +# This is mildly tricky: This script runs itself recursively. The
> > +# first time it is run it does not take the if-branch below and
> > +# executes on the host computer. It uses the builder.sh script to run
> > +# itself inside of docker passing "docker" as the argument causing the
> > +# commands in the if-branch to be executed within the docker
> > +# container.
> > +if [ "${1-}" = "docker" ]; then
> > +    time make STATIC=1 build GOFLAGS="-race"
> > +    check_static cockroach
> > +    strip -S cockroach
>
> nope. I've been using stripped binaries everywhere, and pprof is perfectly
> happy.
>
> Given the brokenness of -s, and the fact that -w doesn't do much, I'll
> stick with strip for now.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/6228/files/562bb54f86997052aa93b63f5c3104e92df9270e#r60713289>
>
-- 

-- Tobias